instrumentation/openai-client-instrumentation/src/test/java/co/elastic/otel/openai/ChatTest.java

```diff
@@ -0,0 +1,2387 @@
+    double nanosPerSecond = Duration.ofSeconds(1).toNanos();
+    return point -> assertThat(point.getSum()).isStrictlyBetween(0.0, toNanos / nanosPerSecond);
+  }
+
+  private static void assertThatValueIsEmptyMap(Value<?> value) {
+    assertThat(value.getValue()).isInstanceOfSatisfying(List.class, l -> assertThat(l).isEmpty());
+  }
+
+  private static void assertThatChoiceBodyHasNoContent(Value<?> body, String finishReason) {
+    ValAssert.map().entry("index", 0).entry("finish_reason", finishReason).accept("", body);
+    Value<?> message =
+        ((List<KeyValue>) body.getValue())
+            .stream().filter(kv -> kv.getKey().equals("message")).findFirst().get().getValue();
+    assertThatValueIsEmptyMap(message);
+  }
+
+  @Test
+  void chat() throws Exception {
+    InstrumentationSettingsAccessor.setCaptureMessageContent(openai.client, false);
+
+    ChatCompletionCreateParams params =
+        ChatCompletionCreateParams.builder()
+            .messages(Collections.singletonList(createUserMessage(TEST_CHAT_INPUT)))
+            .model(TEST_CHAT_MODEL)
+            .build();
+
+    long startTimeNanos = System.nanoTime();
+    ChatCompletion chatCompletion = openai.client.chat().completions().create(params);
+    long durationNanos = System.nanoTime() - startTimeNanos;
+    chatCompletion.validate();
+
+    List<SpanData> spans = testing.spans();
+    assertThat(spans)
+        .hasSize(1)
+        .first()
+        .satisfies(
+            span -> {
+              assertThat(span)
+                  .hasAttribute(GEN_AI_OPERATION_NAME, "chat")
+                  .hasAttribute(GEN_AI_REQUEST_MODEL, TEST_CHAT_MODEL)
+                  .hasAttribute(GEN_AI_SYSTEM, "openai")
+                  .hasAttributesSatisfying(
+                      att -> assertThat(att.get(GEN_AI_RESPONSE_ID)).startsWith("chatcmpl-"))
+                  .hasAttribute(GEN_AI_RESPONSE_MODEL, TEST_CHAT_RESPONSE_MODEL)
+                  .hasAttributesSatisfying(
+                      attribs -> {
+                        assertThat(attribs.get(GEN_AI_RESPONSE_FINISH_REASONS))
+                            .containsExactly("stop");
+                      })
+                  .hasAttribute(GEN_AI_USAGE_INPUT_TOKENS, 22L)
+                  .hasAttribute(GEN_AI_USAGE_OUTPUT_TOKENS, 3L)
+                  .hasAttribute(SERVER_ADDRESS, "localhost")
+                  .hasAttribute(SERVER_PORT, (long) openai.getPort());
+            });
+
+    SpanContext spanCtx = spans.get(0).getSpanContext();
+    assertThat(testing.logRecords())
+        .hasSize(2)
+        .anySatisfy(
+            log -> {
+              assertThat(log)
+                  .hasAttributesSatisfying(
+                      attr ->
+                          assertThat(attr)
+                              .containsEntry(GEN_AI_SYSTEM, "openai")
+                              .containsEntry("event.name", "gen_ai.user.message"))
```

Comment:
[question] does values of `event.name` are part of the semantic conventions ?

Reply:
They are part of the semantic conventions, but not yet of the auto-generated libs it seems.